bindings/gumjs/gumquickprocess.c

```diff
@@ -622,3 +635,61 @@ gum_quick_exception_handler_on_exception (GumExceptionDetails * details,
 
   return handled;
 }
+
+GUMJS_DEFINE_FUNCTION (gumjs_process_run_on_thread)
+{
+  GumQuickScope scope = GUM_QUICK_SCOPE_INIT (core);
+  GumThreadId thread_id;
+  JSValue user_func;
+  GumQuickRunOnThreadContext sync_ctx;
+  GumStalker * stalker;
+  gboolean success;
+
+  if (!_gum_quick_args_parse (args, "ZF", &thread_id, &user_func))
+    return JS_EXCEPTION;
+
+  if (thread_id == 0)
+    return JS_UNDEFINED;
```

Comment:
We don't know (at this level) whether a thread ID of zero is valid, since it's up to the OS.

Reply:
Removed, also for v8.